kernel/interrupts/src/x86_64/mod.rs

```diff
@@ -32,6 +32,19 @@ static RESERVED_IRQ_LIST: [u8; 3] = [
 ];
 
 
+#[macro_export]
+macro_rules! interrupt_handler {
```

Comment:
add some docs with an explanation of each macro argument and also some example use cases. Just by looking at the macro definition itself, you wouldn't be able to tell that the `$x86_64_interrupt_number` needs to be an `Option<u8>`.

Reply:
Here's the doc I added (same for both archs):
![image](https://user-images.githubusercontent.com/61582713/234931188-4b8fe98d-1911-43f6-a612-9229552bfdfb.png)

(hi res screenshot)